static/docs/commands-reference/pull.md

```diff
@@ -1,53 +1,123 @@
 # pull
 
-Pulls data files to the working space.
+Pulls data files to the local workspace from a remote workspace.
 
-The set of data files to pull (usually it means downloading from the remote
-storage if file is not in the local cache yet) is determined by analyzing all
-`.dvc` files in the current branch, unless `--all-branches` is specified.
+## Usage
 
-The command pulls only output of a specific stage if dvc file is specified `dvc pull data.zip.dvc`.
+```usage
+    usage: dvc pull [-h] [-q | -v] [-j JOBS]
+                    [--show-checksums] [-r REMOTE]
+                    [-a] [-T] [-d] [-f] [-R]
+                    [targets [targets ...]]
+
+    Pull data files from the cloud.
+
+    positional arguments:
+      targets               DVC files.
+```
+
+## Description
+
+DVC supports sharing a data cache between workspaces.  Members of a team may
+be working with the same data, but using different algorithms, and will find
+it useful to share data.  The `dvc pull` command is analogous to `git pull`,
+and allows one to retrieve data from a remote cache.
+
+Specifically, the `dvc pull` command downloads files from a remote cache based
+on which have been updated in, or are new to, the remote cache.  By definition
+the `dvc pull` command will always interact between a local and remote cache.
+
+For an overview of the process of sharing data between DVC workspaces and remote
+caches, see:
+[Share Data And Model Files](/doc/use-cases/share-data-and-model-files)
+
+Using the command `dvc status --remote REMOTE` determines whether the local
+cache has differences from the named remote cache.
+
+The value for `REMOTE` is a cache name defined using the `dvc remote` command.
```

Comment:
What happens if no remote is defined? Does it just behave as `dvc checkout` then?

Reply:
I've added a sentence that an error message is printed.

Reply:
No, I mean if there are no remotes in the repository at all. No default remote, nothing. How will `dvc pull` behave in this case?

Reply:
As expected if no remotes are configured at all a different error message is printed.  The text now says so for both cases.